jerry-core/ecma/builtin-objects/ecma-builtin-array-prototype.c

```diff
@@ -1218,10 +1079,12 @@ ecma_builtin_array_prototype_object_sort (ecma_value_t this_arg, /**< this argum
   /* Sorting. */
   if (copied_num > 1 && ecma_is_value_empty (ret_value))
   {
+    const sort_compare_t sort_cb = &ecma_builtin_array_prototype_object_sort_compare_helper;
     ECMA_TRY_CATCH (sort_value,
-                    ecma_builtin_array_prototype_object_array_heap_sort_helper (values_buffer,
-                                                                                (int)(copied_num - 1),
-                                                                                arg1),
+                    ecma_builtin_helper_array_heap_sort_helper (values_buffer,
+                                                                (int)(copied_num - 1),
```

Comment:
There should be a space after casting. It is strange that style checkers don't catch it. See your own `(double)` castings below. There are multiple instances of this style issue, please fix all.

I am thinking about the int type here. ES allows the full uint32 range for array indicies. However we cannot allocate a local buffer that big, so it might be not possible to use that big numbers anyway. What is your opinion?

Reply:
Wouldn't hurt to be consistent with ES and "allow" the full range even if we have hardware limitations, IMHO.